storage/StorageImpl.cc

```diff
@@ -46,16 +46,46 @@
 namespace storage
 {
 
-    Storage::Impl::Impl(const Storage& storage, const Environment& environment)
-	: storage(storage), environment(environment), arch(false),
-	  default_mount_by(MountByType::UUID), tmp_dir("libstorage-XXXXXX")
+    Storage::Impl::Impl(const Storage& storage, const Environment& environment,
+			const ActivateCallbacks* activate_callbacks)
+	: storage(storage), environment(environment), activate_callbacks(activate_callbacks),
```

Comment:
So far yes since activation must happen before probing which is done in the constructor.

In Storage.h I have already added a not to change the interface by introducing a probe function and not probing in the constructor. I currently working on that.